control-plane-operator/controllers/hostedcontrolplane/oauth/route.go

```diff
@@ -6,16 +6,19 @@ import (
 	hyperv1 "github.com/openshift/hypershift/api/v1alpha1"
 	"github.com/openshift/hypershift/control-plane-operator/controllers/hostedcontrolplane/manifests"
 	"github.com/openshift/hypershift/support/config"
+	"github.com/openshift/hypershift/support/util"
 )
 
-func ReconcileRoute(route *routev1.Route, ownerRef config.OwnerRef, strategy *hyperv1.ServicePublishingStrategy) error {
+func ReconcileRoute(route *routev1.Route, ownerRef config.OwnerRef, strategy *hyperv1.ServicePublishingStrategy, defaultIngressDomain string) error {
 	ownerRef.ApplyTo(route)
 	if strategy.Route != nil && strategy.Route.Hostname != "" {
 		if route.Annotations == nil {
 			route.Annotations = map[string]string{}
 		}
 		route.Annotations[hyperv1.ExternalDNSHostnameAnnotation] = strategy.Route.Hostname
 		route.Spec.Host = strategy.Route.Hostname
+	} else {
+		route.Spec.Host = util.ShortenRouteHostnameIfNeeded(route.Name, route.Namespace, defaultIngressDomain)
```

Comment:
not a problem with this PR but... where is the `Private` handling here? :thinking:

Reply:
I guess we haven't so far tried to use oauth in a private setup? Seems like we would need to handle it